src/to-have-style.ts

```diff
@@ -1,56 +1,48 @@
+import { StyleSheet } from 'react-native';
+import type { ImageStyle, StyleProp, TextStyle, ViewStyle } from 'react-native';
+import type { ReactTestInstance } from 'react-test-renderer';
 import { matcherHint } from 'jest-matcher-utils';
 import { diff } from 'jest-diff';
 import chalk from 'chalk';
-import type { ReactTestInstance } from 'react-test-renderer';
-import type { ImageStyle, StyleProp, TextStyle, ViewStyle } from 'react-native';
-import { StyleSheet } from 'react-native';
-import { checkReactElement, display } from './utils';
+import { checkReactElement } from './utils';
 
 type Style = TextStyle | ViewStyle | ImageStyle;
+type StyleLike = Record<string, unknown>;
```

Comment:
why the styleLike? passing an undefined as a styleProp is valid but the flatten won't return a Record

Reply:
Thanks for pointing out potential `undefined` return from `StyleSheet.flatten()` I've added check for that.

Regarding why `Record` and not concrete `Style`, it has to do with `Object.keys()` not handling well `Style`, so we  either need to we sprinkle `Object.keys` or `StyleSheet.flatten` with `as` typecasts. Since flattened style is essential `Record<string, any>` or `undefined`, then it makes sense to use `Record` as it does not quire additional typecasts in the checking functions downstream.

Reply:
Just FYI, that `StyleSheet.flatten` may be deprecated in some far future: https://github.com/react-native-community/discussions-and-proposals/pull/496/files#diff-8e74bede0f5d8ed8c232710e7e2fef8af85fdb5fc65c23bcb415e4e6beafb518R310